contracts/token/ERC20/extensions/ERC7540.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {IERC7540} from "../../../interfaces/IERC7540.sol";
+import {IERC20} from "../../../interfaces/IERC20.sol";
+import {ERC4626} from "./ERC4626.sol";
+import {SafeERC20} from "../utils/SafeERC20.sol";
+import {Math} from "../../../utils/math/Math.sol";
+import {ReentrancyGuard} from "../../../utils/ReentrancyGuard.sol";
+
+/**
+ * @dev Abstract implementation of the ERC-7540 standard, extending ERC-4626.
+ */
+abstract contract ERC7540 is ERC4626, IERC7540, ReentrancyGuard {
```

Comment:
Is reentrancy guard really necessary? None of the contract in the codebase use it. We always make sure that behaviour is safe without it.

Reply:
Makes sense, i think its an overhead, so i agree with your suggestion to remove it. @Amxx

Reply:
Updated